pkg/operator/controller/controller.go

```diff
@@ -90,6 +91,9 @@ type Config struct {
 // events.
 type reconciler struct {
 	Config
+
+	dnsConfig   *configv1.DNS
+	infraConfig *configv1.Infrastructure
```

Comment:
These should be retrieved from the client/cache every reconcile so we can do change detection.